homeassistant/components/obihai/config_flow.py

```diff
@@ -40,17 +59,11 @@ async def async_step_user(
 
         if user_input is not None:
             self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
-            if await self.hass.async_add_executor_job(
-                validate_auth,
-                user_input[CONF_HOST],
-                user_input[CONF_USERNAME],
-                user_input[CONF_PASSWORD],
-            ):
+            if not (errors := await async_credential_errors(self.hass, user_input)):
```

Comment:
I'm still not a great fan of creating the error dictionary inside the function.
I would still prefer something like this:
```python
    if await async_credential_errors(self.hass, user_input):
        return self.async_create_entry(...)
    errors["base"] = "cannot_connect"
```

Reply:
Can't blame me for trying one more time... :)